dynomitemanager-core/src/main/java/com/netflix/dynomitemanager/storage/RedisStorageProxy.java

```diff
@@ -205,6 +218,9 @@ public boolean takeSnapshot() {
                 throw e;
             }
             logger.warn("Redis: There is already a pending BGREWRITEAOF/BGSAVE.");
+        } catch (JedisConnectionException e) {
+            logger.error("Redis: BGREWRITEAOF/BGSAVE cannot be completed");
+            localRedisDisconnect();
         }
```

Comment:
if you disconnect, what happens to the localJedis at line 231

Reply:
Since we already are in a bad state of BGREWRITEOF we should not continue.